vignettes/syntax_extractor.Rmd

```diff
@@ -0,0 +1,305 @@
+---
+title: "Syntax Extractor"
+output: 
+  html_document:
+    toc: true
+    number_sections: true
+vignette: >
+  %\VignetteEngine{knitr::rmarkdown}
+  %\VignetteIndexEntry{Kaiaulu Syntax Extractor}
+  %\VignetteEncoding{UTF-8}
+---
+
+```{r eval=FALSE}
+rm(list = ls())
+seed <- 1
+set.seed(seed)
+
+# Load libraries
+  require(kaiaulu)
+  require(data.table)
+  require(yaml)
+  require(stringi)
+  require(XML)
+  require(gt)
+```
+
+# Introduction
+
+In open-source projects, code is often spread across many files and organized in various ways. Understanding this code at a glance can be difficult, especially when projects grow large and complex. This is where syntax extraction comes in.
```

Comment:
I would remove this. The motivation fits better to why someone would need Kaiaulu than the feature you are working here!